beacon-chain/blockchain/receive_blob.go

```diff
@@ -0,0 +1,23 @@
+package blockchain
+
+import "github.com/prysmaticlabs/prysm/v4/container/slice"
+
+// SendNewBlobEvent sends a message to the BlobNotifier channel that the blob
+// for the blocroot `root` is ready in the database
+func (s *Service) SendNewBlobEvent(root [32]byte, index uint64) {
+	s.blobNotifier.Lock()
+	defer s.blobNotifier.Unlock()
```

Comment:
you're holding the lock until the read happens, which will deadlock the reader that needs to hold the lock before it gets to the channel read code.